src/Microsoft.AspNetCore.Sockets/Internal/SocketEventSource.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Sockets.Internal
+{
+    [EventSource(Name = "Microsoft-AspNetCore-Sockets")]
+    internal class SocketEventSource : EventSource
+    {
+        public static readonly SocketEventSource Log = new SocketEventSource();
+
+        private readonly EventCounter _connectionsStarted;
+        private readonly EventCounter _connectionsStopped;
+        private readonly EventCounter _connectionsTimedOut;
+        private readonly EventCounter _connectionDuration;
+
+        private SocketEventSource()
+        {
+            _connectionsStarted = new EventCounter("ConnectionsStarted", this);
+            _connectionsStopped = new EventCounter("ConnectionsStopped", this);
+            _connectionsTimedOut = new EventCounter("ConnectionsTimedOut", this);
+            _connectionDuration = new EventCounter("ConnectionDuration", this);
+        }
+
+        // This has to go through NonEvent because only Primitive types are allowed
+        // in function parameters for Events
+        [NonEvent]
+        public void ConnectionStop(string connectionId, ValueStopwatch timer)
+        {
+            if (IsEnabled())
+            {
+                var duration = timer.IsActive ? timer.GetElapsedTime().TotalMilliseconds : 0.0;
+                _connectionDuration.WriteMetric((float)duration);
+                _connectionsStopped.WriteMetric(1.0f);
+
+                if (IsEnabled(EventLevel.Informational, EventKeywords.None))
+                {
+                    ConnectionStop(connectionId);
+                }
+            }
+        }
+
+        [Event(eventId: 1, Level = EventLevel.Informational, Message = "Started connection '{0}'.")]
+        public ValueStopwatch ConnectionStart(string connectionId)
```

Comment:
Any other values we could add to the payload? We don't know the transport yet, correct? Maybe include the request's TraceIdentifier if we have it?

Reply:
We don't have any info at this point